File: estudiantes-backend/src/main/java/com/example/demo/model/Estudiante.java
package com.example.demo.model;

public class Estudiante {
    private Long id;
    private String nombre;
    private String apellido;
    private int edad;
    private String curso;

    public Estudiante() {
    }

    public Estudiante(Long id, String nombre, String apellido, int edad, String curso) {
        this.id = id;
        this.nombre = nombre;
        this.apellido = apellido;
        this.edad = edad;
        this.curso = curso;
    }

    // Getters y Setters
    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public String getApellido() {
        return apellido;
    }

    public void setApellido(String apellido) {
        this.apellido = apellido;
    }

    public int getEdad() {
        return edad;
    }

    public void setEdad(int edad) {
        this.edad = edad;
    }

    public String getCurso() {
        return curso;
    }

    public void setCurso(String curso) {
        this.curso = curso;
    }
}

File: estudiantes_flutter/lib/models/estudiante.dart
class Estudiante {
  int? id;
  String nombre;
  String apellido;
  int edad;
  String curso;

  Estudiante({
    this.id,
    required this.nombre,
    required this.apellido,
    required this.edad,
    required this.curso,
  });

  factory Estudiante.fromJson(Map<String, dynamic> json) {
    return Estudiante(
      id: json['id'],
      nombre: json['nombre'],
      apellido: json['apellido'],
      edad: json['edad'],
      curso: json['curso'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'nombre': nombre,
      'apellido': apellido,
      'edad': edad,
      'curso': curso,
    };
  }
}